### PR TITLE
Creating a save_as function in workspace with path checking.

### PR DESCRIPTION
**Pull request by MatthieuCMira**
**GEOPY-993 - Add save_as option to create a new file **
Replace "save" functions to "save_as" everywhere

Add Warning deprecation to save functions in both workspace and entity.

PROBLEM: save_as function in entity does not allow the creation of a dataframe. maybe remove savee and save_as in entity?

### CODE REVIEW

File: geoh5py/shared/entity.py [hunk 1]
```diff
@@ -515,11 +518,29 @@ def remove_data_from_group(
...
+    #     :param filepath: The path of the new workspace.
+    #     :param add_children: Option to also save the children.
+    #     :return: the entity in the new workspace
+    #     """
+    #     # create a new workspace
+    #     workspace = Workspace.create(filepath)
+    #
+    #     # save the entity in the new workspace
+    #     return workspace.save_entity(self, add_children=add_children)
+
```
Comment (domfournier) [inline geoh5py/shared/entity.py:543]: Ok well, looks like that's not possible without a major re-write of the class. Let's just remove save_as here then...

File: tests/copy_entity_test.py [hunk 2]
```diff
@@ -88,9 +88,14 @@ def test_copy_entity(tmp_path: Path):
         with pytest.raises(AssertionError, match="Input metadata must be of type"):
             entity.metadata = 0
 
-        _ = entity.save()
+        # _ = entity.save_as(tmp_path / r"testProject2.geoh5")
```
Comment (domfournier) [inline tests/copy_entity_test.py:91]:
```suggestion
```

**Review by domfournier — APPROVED** [review]
Pretty simple if we keep it on the workspace. We can re-visit this later for the entity.

File: geoh5py/shared/entity.py (hunk 1 above)
Comment (MatthieuCMira) [reply geoh5py/shared/entity.py:543]: removing the commented part of the code.

File: geoh5py/shared/entity.py [hunk 3]
```diff
@@ -28,6 +29,8 @@
 
 from geoh5py.shared.utils import str2uuid
 
+# from geoh5py.workspace import Workspace
+
```
Comment (domfournier) [inline geoh5py/shared/entity.py:33]:
```suggestion
```

**Review by domfournier — DISMISSED** [review]

**Review by domfournier — APPROVED** [review]
Great, that worked!

**Pull request merged**